### PR TITLE
PWGHF: add info about K-pi pairs from Lc->pKpi decays, and add direct/reso. decay channel info

### PR DESCRIPTION
**Pull request by mfaggin**
1. add possibility to cu on K-pi invariant mass
2. propagate the info in the tree creator and polarization task
3. add info about direct-reso. decay channel in tree creator and polarization task

### CODE REVIEW

File: PWGHF/TableProducer/candidateSelectorLc.cxx [hunk 1]
```diff
@@ -235,6 +238,14 @@ struct HfCandidateSelectorLc {
       }
     }
 
+    /// cut on the Kpi pair invariant mass, to study Lc->pK*(->Kpi)
+    const double cutMassKPi = cuts->get(pTBin, "mass (Kpi)");
+    if (cutMassKPi > 0) {
+      if (std::abs(RecoDecay::m(std::array{trackKaon.pVector(), trackPion.pVector()}, std::array{o2::constants::physics::MassKaonCharged, o2::constants::physics::MassPiPlus}) - massK0Star892) > cutMassKPi) {
```
Comment (fgrosa) [inline PWGHF/TableProducer/candidateSelectorLc.cxx:244]: I would add a function in the `hfHelper` for the massKPi calculation, as for the massPKPi, similar to those for the massKK of the Ds https://github.com/AliceO2Group/O2Physics/blob/d3b76b374b0aa86092f1bf70adf77eecffbc7b63/PWGHF/Core/HfHelper.h#L245-L255

File: PWGHF/TableProducer/treeCreatorLcToPKPi.cxx [hunk 2]
```diff
@@ -477,13 +486,15 @@ struct HfTreeCreatorLcToPKPi {
...
+              candidate.flagMcDecayChanRec(),
+              FunctionInvMassKPi);
           }
         }
       };
 
-      fillTable(0, candidate.isSelLcToPKPi(), hfHelper.invMassLcToPKPi(candidate), hfHelper.ctLc(candidate), hfHelper.yLc(candidate), hfHelper.eLc(candidate));
-      fillTable(1, candidate.isSelLcToPiKP(), hfHelper.invMassLcToPiKP(candidate), hfHelper.ctLc(candidate), hfHelper.yLc(candidate), hfHelper.eLc(candidate));
+      fillTable(0, candidate.isSelLcToPKPi(), hfHelper.invMassLcToPKPi(candidate), hfHelper.ctLc(candidate), hfHelper.yLc(candidate), hfHelper.eLc(candidate), RecoDecay::m(std::array{trackNeg.pVector(), trackPos2.pVector()}, std::array{o2::constants::physics::MassKaonCharged, o2::constants::physics::MassPiPlus}));
+      fillTable(1, candidate.isSelLcToPiKP(), hfHelper.invMassLcToPiKP(candidate), hfHelper.ctLc(candidate), hfHelper.yLc(candidate), hfHelper.eLc(candidate), RecoDecay::m(std::array{trackNeg.pVector(), trackPos1.pVector()}, std::array{o2::constants::physics::MassKaonCharged, o2::constants::physics::MassPiPlus}));
```
Comment (fgrosa) [inline PWGHF/TableProducer/treeCreatorLcToPKPi.cxx:497]: See comment above about the function in the `hfHelper`

File: PWGHF/TableProducer/treeCreatorLcToPKPi.cxx [hunk 3]
```diff
@@ -731,13 +745,15 @@ struct HfTreeCreatorLcToPKPi {
...
+              -1,
+              FunctionInvMassKPi);
           }
         }
       };
 
-      fillTable(0, candidate.isSelLcToPKPi(), hfHelper.invMassLcToPKPi(candidate), hfHelper.ctLc(candidate), hfHelper.yLc(candidate), hfHelper.eLc(candidate));
-      fillTable(1, candidate.isSelLcToPiKP(), hfHelper.invMassLcToPiKP(candidate), hfHelper.ctLc(candidate), hfHelper.yLc(candidate), hfHelper.eLc(candidate));
+      fillTable(0, candidate.isSelLcToPKPi(), hfHelper.invMassLcToPKPi(candidate), hfHelper.ctLc(candidate), hfHelper.yLc(candidate), hfHelper.eLc(candidate), RecoDecay::m(std::array{trackNeg.pVector(), trackPos2.pVector()}, std::array{o2::constants::physics::MassKaonCharged, o2::constants::physics::MassPiPlus}));
+      fillTable(1, candidate.isSelLcToPiKP(), hfHelper.invMassLcToPiKP(candidate), hfHelper.ctLc(candidate), hfHelper.yLc(candidate), hfHelper.eLc(candidate), RecoDecay::m(std::array{trackNeg.pVector(), trackPos1.pVector()}, std::array{o2::constants::physics::MassKaonCharged, o2::constants::physics::MassPiPlus}));
```
Comment (fgrosa) [inline PWGHF/TableProducer/treeCreatorLcToPKPi.cxx:756]: See above the comment about the function in the `hfHelper`

File: PWGHF/D2H/Tasks/taskCharmPolarisation.cxx [hunk 4]
```diff
@@ -806,6 +814,8 @@ struct TaskPolarisationCharmHadrons {
               outputMl[2] = candidate.mlProbLcToPKPi()[2];
             }
           }
+          // invariant mass of the KPi pair
+          invMassKPiLc = RecoDecay::m(std::array{candidate.pVectorProng1(), candidate.pVectorProng2()}, std::array{massKaon, massPi});
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskCharmPolarisation.cxx:818]: I would add a function in the `hfHelper` for the massKPi calculation, as for the massPKPi, similar to those for the massKK of the Ds https://github.com/AliceO2Group/O2Physics/blob/d3b76b374b0aa86092f1bf70adf77eecffbc7b63/PWGHF/Core/HfHelper.h#L245-L255

File: PWGHF/D2H/Tasks/taskCharmPolarisation.cxx [hunk 5]
```diff
@@ -830,6 +840,8 @@ struct TaskPolarisationCharmHadrons {
               outputMl[2] = candidate.mlProbLcToPiKP()[2];
             }
           }
+          // invariant mass of the KPi pair
+          invMassKPiLc = RecoDecay::m(std::array{candidate.pVectorProng1(), candidate.pVectorProng0()}, std::array{massKaon, massPi});
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskCharmPolarisation.cxx:844]: See comment above about the function in the `hfHelper`

File: PWGHF/TableProducer/candidateSelectorLc.cxx [hunk 6]
```diff
@@ -101,6 +101,8 @@ struct HfCandidateSelectorLc {
 
   HistogramRegistry registry{"registry"};
 
+  double massK0Star892 = 0.89555; // M(K*0(892)) = 895.55 Mev/c2
```
Comment (fgrosa) [inline PWGHF/TableProducer/candidateSelectorLc.cxx:104]: I added the K* mass to the physics constants in O2, let's remember to modify this hard-coded value once https://github.com/AliceO2Group/AliceO2/pull/13235 gets merged

**Review by fgrosa — CHANGES_REQUESTED** [review]
Hi @mfaggin thanks! I basically have only one comment, see below.

File: PWGHF/TableProducer/candidateSelectorLc.cxx (hunk 6 above)
Comment (mfaggin) [reply PWGHF/TableProducer/candidateSelectorLc.cxx:104]: good to know, thank you

**Comment by mfaggin** [comment]
thanks @fgrosa I've implemented your suggestions

**Comment by fgrosa** [comment]
> thanks @fgrosa I've implemented your suggestions

Hi @mfaggin thanks! Since https://github.com/AliceO2Group/AliceO2/pull/13235 was merged, we could either merge yours now and implement it tomorrow, or you include it in this PR and we wait for tomorrow (no preference from my side).

**Comment by mfaggin** [comment]
> > thanks @fgrosa I've implemented your suggestions
> 
> Hi @mfaggin thanks! Since [AliceO2Group/AliceO2#13235](https://github.com/AliceO2Group/AliceO2/pull/13235) was merged, we could either merge yours now and implement it tomorrow, or you include it in this PR and we wait for tomorrow (no preference from my side).

Hi @fgrosa thank you. Let's put everything here. I'm going to do the commit now.

**Pull request merged**